src/test/harness.rs

```diff
@@ -187,16 +183,26 @@ impl ExpectedMessage {
     }
 }
 
-pub fn expect_messages(results: LsResultList, expected: &[&ExpectedMessage]) {
-    let start_clock = SystemTime::now();
-    let mut results_count = results.lock().unwrap().len();
-    while results_count < expected.len() {
-        if start_clock.elapsed().unwrap().as_secs() >= TEST_TIMEOUT_IN_SEC {
-            panic!("Hit timeout");
+macro_rules! wait_for_n_results {
+    ($n:expr, $results:expr) => {{
+        use std::time::{Duration, SystemTime};
+        use std::thread;
+
+        let timeout = Duration::from_secs(320);
```

Comment:
Could you leave the magic number as a const please?

Reply:
There are two things I don't like with magic numbers, not knowing what the number means and not knowing what units the number is in. In this case the it's actually clearer than before, by looking at the source we can see its a timeout for waiting for the _n_ results, rather than a timeout for the entire test suite, or each test etc.

It's also only used here, and _here_ is now a macro which we intend to be used in another file. So a const is less helpful.

Actually the problem I have with this bit is **320 seconds** is far too long for a test timeout, but I try not to sneakily change too much when I'm moving stuff around.

Reply:
Another benefit is simply having 'configuration options' easily discoverable and tweakable. It's also an indicator that there *is* an option that is basically arbitrary and might be worth fiddling with (and making that discoverable without getting stuck into the source code).

The idea with this timeout is that it should never, ever be hit, even when everything goes wrong, which is why it is so long. Given that it is so long, I'm kinda doubtful about its utility at all, but that is an issue for another time.

Reply:
For _use once_ scenarios I guess I prefer self-documenting code inline. This really isn't a big deal though, as macro's can't reference consts as easily I think it has the edge. If you'd prefer to lose the macro, even though they're beneficial for test code, we can go back to the const.